microcosm_fastapi/logging_data_map.py

```diff
@@ -0,0 +1,52 @@
+"""
+Used to store information that useful for audit logging purposes
+
+"""
+from typing import Optional
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class LoggingInfo:
+    operation_name: Optional[str] = None
+    function_name: Optional[str] = None
+
+
+class LoggingDataMap:
+    def __init__(self):
+        self.data_map = {}
+
+    def add_entry(self, namespace, operation, function_name):
```

Comment:
Let's add type annotations for this method and all other.

Reply:
Done